src/twitter.d.ts

```diff
@@ -0,0 +1,27 @@
+declare module "twitter-v2" {
+    export interface Credentials {
+        consumer_key?: string;
+        consumer_secret?: string;
+        bearer_token?: string;
+        access_token_key?: string;
+        access_token_secret?: string;
+    }
+
+    export interface TwitterResponse<T extends any> {
+        data: T;
+    }
+
+    export default class Twitter {
+        constructor(credentials: Credentials);
+        public get<T>(
+            endpoint: string,
+            parameters?: any
+        ): Promise<TwitterResponse<T>>;
+        public post<T>(
+            endpoint: string,
+            body?: any,
+            parameters?: any
+        ): Promise<TwitterResponse<T>>;
+        public stream(endpoint: string, parameters?: any): any;
```

Comment:
Interesting. We probably want to also define `TwitterStream` as a class. It's going to be a little funky thanks to the async iterator but it looks like there's a way to express this

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-3.html